dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/internal/ZipFileContainer.java

```diff
@@ -902,6 +921,25 @@ public int locatePath(String r_path) {
         }
         return location;
     }
+    
+    @Trivial
+    private int locateMultiReleasePath(String path) {
+        // Multi-release files may be at /com/foo/A.class and /META-INF/versions/9/com/foo/A.class
+        // Since we commonly blindly try-load META-INF and java.* entries, optimize those out
+        if (path == null || !path.endsWith(".class") || path.startsWith("META-INF") || path.startsWith("java/")) {
```

Comment:
!path.endsWith(".class") should be removed.  Non class files can also be Multiple-Releaseable.